spring-grpc-core/src/main/java/org/springframework/grpc/client/DefaultGrpcChannelFactory.java

```diff
@@ -70,6 +70,18 @@ public DefaultGrpcChannelFactory(List<GrpcChannelBuilderCustomizer<T>> globalCus
 		this.interceptorsConfigurer = interceptorsConfigurer;
 	}
 
+	/**
+	 * Whether this factory supports the given target string. The target can be either a
+	 * valid nameresolver-compliant URI, an authority string as described in
+	 * {@link Grpc#newChannelBuilder(String, ChannelCredentials)}.
+	 * @param target the target string as described in method javadocs
+	 * @return true unless the target begins with 'in-process:'
+	 */
+	@Override
+	public boolean supports(String target) {
+		return !target.startsWith("in-process:");
```

Comment:
All "default" channel factories (including those users may specialize from) support by default. I am not super excited about bleeding this here, however, it is a string and nothing more.